dpnp/dpnp_array.py

```diff
@@ -1000,15 +993,61 @@ def take(self, indices, axis=None, out=None, mode='raise'):
 
     def transpose(self, *axes):
         """
-        Returns a view of the array with axes permuted.
+        Returns a view of the array with axes transposed.
 
-        .. seealso::
-           :obj:`dpnp.transpose` for full documentation,
-           :meth:`numpy.ndarray.reshape`
+        For full documentation refer to :obj:`numpy.ndarray.transpose`.
+
+        Returns
+        -------
+        y : dpnp.ndarray
+            View of the array with its axes suitably permuted.
 
+        See Also
+        --------
+            :obj:`dpnp.transpose` : Equivalent function.
+            :obj:`dpnp.ndarray.ndarray.T` : Array property returning the array transposed.
+            :obj:`dpnp.ndarray.reshape` : Give a new shape to an array without changing its data.
+
+        Examples
+        --------
+        >>> import dpnp as dp
+        >>> a = dp.array([[1, 2], [3, 4]])
+        >>> a
+        array([[1, 2],
+               [3, 4]])
+        >>> a.transpose()
+        array([[1, 3],
+               [2, 4]])
+        >>> a.transpose((1, 0))
+        array([[1, 3],
+               [2, 4]])
+        >>> a.transpose(1, 0)
+        array([[1, 3],
+               [2, 4]])
+
+        >>> a = dp.array([1, 2, 3, 4])
+        >>> a
+        array([1, 2, 3, 4])
+        >>> a.transpose()
+        array([1, 2, 3, 4])
+            
         """
 
-        return dpnp.transpose(self, axes)
+        ndim = self.ndim
+        if ndim < 2:
+            return self
+
+        res = self.__new__(dpnp_array)
+        if ndim == 2:
+            res._array_obj = self._array_obj.T
+        else:
+            if len(axes) == 0 or len(axes) == 1 and axes[0] is None:
+                # self.transpose().shape == self.shape[::-1]
+                # self.transpose(None).shape == self.shape[::-1]
+                axes = tuple((ndim - x - 1) for x in range(ndim))
+
+            res._array_obj = dpt.permute_dims(self._array_obj, axes)
```

Comment:
I get the following error
`a_dpnp = dpnp.ones((1, 2, 3))`
`a_numpy = numpy.ones((1,2,3))`

`numpy.transpose(a_numpy, (1,0,2)).shape`
 `(2, 1, 3)`
 `dpnp.transpose(a_dpnp, (1,0,2)).shape`
 `(2, 1, 3)`

`a_numpy.transpose((1,0,2)).shape`
 `(2, 1, 3)`

`a_dpnp.transpose((1,0,2)).shape`

File /localdisk/work/vtavanas/repos/dpnp/dpnp/dpnp_array.py:1049, in dpnp_array.transpose(self, *axes)
   1044     if len(axes) == 0 or len(axes) == 1 and axes[0] is None:
   1045         # self.transpose().shape == self.shape[::-1]
   1046         # self.transpose(None).shape == self.shape[::-1]
   1047         axes = tuple((ndim - x - 1) for x in range(ndim))
-> 1049     res._array_obj = dpt.permute_dims(self._array_obj, axes)
   1050 return res

File /localdisk/work/vtavanas/repos/dpctl/dpctl/tensor/_manipulation_functions.py:204, in permute_dims(X, axes)
    202 if not isinstance(X, dpt.usm_ndarray):
    203     raise TypeError(f"Expected usm_ndarray type, got {type(X)}.")
--> 204 axes = normalize_axis_tuple(axes, X.ndim, "axes")
    205 if not X.ndim == len(axes):
    206     raise ValueError(
    207         "The length of the passed axes does not match "
    208         "to the number of usm_ndarray dimensions."
    209     )

File /localdisk/work/vtavanas/miniconda3/envs/dpctl_dev/lib/python3.10/site-packages/numpy/core/numeric.py:1397, in normalize_axis_tuple(axis, ndim, argname, allow_duplicate)
   1395         pass
   1396 # Going via an iterator directly is slower than via list comprehension.
-> 1397 axis = tuple([normalize_axis_index(ax, ndim, argname) for ax in axis])
   1398 if not allow_duplicate and len(set(axis)) != len(axis):
   1399     if argname:

File /localdisk/work/vtavanas/miniconda3/envs/dpctl_dev/lib/python3.10/site-packages/numpy/core/numeric.py:1397, in <listcomp>(.0)
   1395         pass
   1396 # Going via an iterator directly is slower than via list comprehension.
-> 1397 axis = tuple([normalize_axis_index(ax, ndim, argname) for ax in axis])
   1398 if not allow_duplicate and len(set(axis)) != len(axis):
   1399     if argname:

TypeError: 'tuple' object cannot be interpreted as an integer

Reply:
thank you for noticing that, now it should work